### ISSUE TITLE
Library blocks if there is no HX711 present

### ISSUE THREAD

**Issue by pearson** [mpibpc-mroose/hx711#2]
* HX711 version: 1.1.2.2
* Python version: 3.5.3
* Operating System: Raspbian Stretch Lite June 2018

### Description

The library blocks if there is no HX711 present. This seems to boil down to a couple of problems:

1. Creating the initial HX711 object will return a valid object even if there is no HX711 connected to the pins.
2. The `reset()` method blocks and does not throw an error if the `_set_channel_gain() fails`. It will log the warning "`WARNING:root:setting gain and channel took more than 60µs`" until Python itself throws a "`RecursionError: maximum recursion depth exceeded in comparison`"

It would be useful for the library to throw errors so that a failed HX711 (for whatever reason) can be detected and handled in some way, rather than simply crashing the application.

### What I Did
I used the simple example code, but without an HX711 attached. This was to simulate a device with a connected HX711 based scale, where the scale is not plugged in to the main device.

```
...
WARNING:root:setting gain and channel took more than 60µs. Time elapsed: 0.0001909989950945601
WARNING:root:setting gain and channel took more than 60µs. Time elapsed: 0.00026499800151214004
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 981, in emit
    msg = self.format(record)
  File "/usr/lib/python3.5/logging/__init__.py", line 831, in format
    return fmt.format(record)
  File "/usr/lib/python3.5/logging/__init__.py", line 569, in format
    if self.usesTime():
  File "/usr/lib/python3.5/logging/__init__.py", line 537, in usesTime
    return self._style.usesTime()
  File "/usr/lib/python3.5/logging/__init__.py", line 381, in usesTime
    return self._fmt.find(self.asctime_search) >= 0
RecursionError: maximum recursion depth exceeded in comparison

```

**Comment by mpibpc-mroose** [comment]
Thanks for your report. The RecursionError came from the Python standard logging module. I tried to prevent that with a small change in how the elapsed time is logged.

The blocking behavior itself is expected. As the HX711 class is some kind of basic blueprint you can and should extend the class. For instance you could easily wrap the reset method of the class with some timeout decorator to achive a non blocking behavior.

If you use https://pypi.org/project/timeout-decorator/ it could look like this:

```
from hx711 import  HX711
import timeout_decorator

class NonBlockingHX711(
    @timeout_decorator(5)
    def _set_channel_gain(self, num):
        return super(NonBlockingHX711, self)._set_channel_gain(num)
```

**Comment by mpibpc-mroose** [comment]
fixed in 7230ff0
